src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoInternalOfT.cs

```diff
@@ -74,6 +75,7 @@ public JsonTypeInfoInternal(JsonSerializerOptions options, JsonObjectInfoValues<
             CreateObjectWithArgs = createObjectWithArgs;
             AddMethodDelegate = addFunc;
             SetCreateObjectFunc(collectionInfo.ObjectCreator);
+            converter.ConfigureJsonTypeInfo(this, Options);
```

Comment:
I feel this should be called as part of construction of the converter

Reply:
Converters in general have a one-to-many relationship with `JsonTypeInfo` instances, so each `JsonTypeInfo` needs to be configured separately.